File: 01-FUNDAMENTOS_DE_PROGRAMACAO/C/Exercicio 22.c
/*
22 - Realizar as funções de fibonachi, fatorial e a somatória de N até 1,  todas com recursividade.
*/

#include<stdio.h>

int fib(int n);
int fatorial(int n);
int soma(int n);

int main ()
{
    int n, resultado, i;
    printf ("Digite o valor de N: ");
    scanf ("%d",&n);

    resultado = fib(n);
    printf ("\nFibonachi: %d",resultado);

    resultado = fatorial(n);
    printf ("\nFatorial: %d\n",resultado);

    resultado = soma(n);
    printf ("\nSoma: %d\n",resultado);


    return 0;
}

int fib(int n)
{
    if (n==0)
        return 0;
    if (n==1)
        return 1;
    return fib(n-1) + fib(n-2);
}

int fatorial(int n)
{
    if (n==0 || n==1)
        return 1;
    return n*fatorial(n-1);
}

int soma(int n)
{
    if (n==1)
        return 1;
    return n + soma(n-1);
}

File: 01-FUNDAMENTOS_DE_PROGRAMACAO/Java/Exercicio22.java
/*
22 - Realizar as funções de fibonachi, fatorial e a somatória de 1 até N,  
todas com recursividade.
 */

import java.util.Scanner;

public class exercicio22 {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        System.out.println("Digite valor de N(inteiro): ");
        int n = teclado.nextInt();
        
        System.out.println("\nFibonachi: "+fib(n));
        System.out.println("\nFatorial: "+fatorial(n));
        
        System.out.println("\nSomatória de 1 até N: "+soma(n));
    }
    
    public static int fib(int n){
        if (n==0) return 0;
        if (n==1) return 1;
        return fib(n-1) + fib(n-2);
    }
    
    public static int fatorial(int n){
        if (n==0 || n==1)
            return 1;
        return n*fatorial(n-1);
    }
    
    public static int soma(int n){
        if (n==1) return 1;
        return n + soma(n-1);
    }
}
